File: backend/app/schemas/cover_letter.py
from datetime import datetime

from pydantic import BaseModel, Field


class CoverLetterContent(BaseModel):
    subject: str | None = None
    body: str = Field(min_length=1)
    notes: str | None = None


class JobCoverLetterResult(CoverLetterContent):
    job_id: int
    generated_at: datetime


class JobCoverLetterRead(BaseModel):
    computed: bool
    cover_letter: JobCoverLetterResult | None = None


class CoverLetterPromptConfig(BaseModel):
    default_system_prompt: str
    custom_system_prompt: str | None = None
    uses_custom: bool


class CoverLetterPromptUpdate(BaseModel):
    system_prompt: str | None = None

File: frontend/src/types/cover_letter.ts
export type JobCoverLetterResult = {
  job_id: number;
  subject: string | null;
  body: string;
  notes: string | null;
  generated_at: string;
};

export type JobCoverLetterRead = {
  computed: boolean;
  cover_letter: JobCoverLetterResult | null;
};

export type CoverLetterPromptConfig = {
  default_system_prompt: string;
  custom_system_prompt: string | null;
  uses_custom: boolean;
};

export type CoverLetterPromptUpdate = {
  system_prompt: string | null;
};

export function formatCoverLetterDate(iso: string): string {
  return new Date(iso).toLocaleString(undefined, {
    dateStyle: "medium",
    timeStyle: "short",
  });
}

export function coverLetterToText(
  subject: string | null,
  body: string,
): string {
  if (subject?.trim()) {
    return `Subject: ${subject.trim()}\n\n${body}`;
  }
  return body;
}
